site/en/docs/extensions/mv3/tut_puppeteer-testing/index.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: "layouts/doc-post.njk"
+title: "Tutorial: Testing Chrome Extensions with Puppeteer"
+seoTitle: "Tutorial: Testing Chrome Extensions with Puppeteer"
+date: 2023-10-09
+description: How to write an automated test for Chrome Etensions in Puppeteer.
+---
+
+_For general advice on testing extensions, see
+[Automated testing for Chrome Extensions][automated-testing]._
+
+[Puppeteer][puppeteer] provides a framework for building automated tests of websites, which also
+includes the ability to test Chrome Extensions. These are high-level integration tests that test the
+functionality of a website or extension once it has been built into the final product.
+
+## Before you start
+
+Clone or download the [chrome-extensions-samples][samples-repo] repository. We'll use the history
+API demo in `api-samples/history/showHistory` as our test extension.
+
+You'll also need to install [Node.JS][node] which is the runtime Puppeteer is built on.
+
+## Writing your test
```

Comment:
Did you consider using Jest here?

Reply:
As a test runner or for unit tests? I think a separate unit testing tutorial is something we could look at in the future, although that was covered at a high level on the other page. I don't think we want to show it as a test runner because this tutorial would quickly explode in complexity as you start needing to explain parallelism, how to setup and tear things down. At least for this first PR I wanted to focus on the bits that are specific to extensions and where good tutorials don't already exist. You could hopefully use this tutorial to set things up in any test runner of your choice.

Reply:
Just as a test runner. Setup / tear down + parallelism are important things we should cover here. If it adds too much overhead I'm fine with starting with this approach.

Reply:
Updated the tutorial with Jest :)